### PR TITLE
switch attribute for checkbox inputs

### LINKED ISSUE #496: Expose <input type=checkbox switch> as role=switch
Once https://github.com/whatwg/html/pull/9546 has agreement and gets merged, that is.

### PR DESCRIPTION
**Pull request by scottaohara**
closes #496 if this feature goes through.

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Oct 18, 2023, 12:44 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [Spec Generator](https://www.w3.org/2015/labs/) - Spec Generator is the web service used to build specs that rely on ReSpec.

:link: [Related URL](https://labs.w3.org/spec-generator/?type=respec&url=https%3A%2F%2Fraw.githubusercontent.com%2Fw3c%2Fhtml-aam%2F326ce09e264057f4e560e17efaa2a725c5f40cf2%2Findex.html%3FisPreview%3Dtrue)

```

😭  Sorry, there was an error generating the HTML. Please report this issue!
Specification: http://labs.w3.org/spec-generator/uploads/FD9Dh7/index.html?isPreview=true%3FisPreview%3Dtrue&publishDate=2023-10-18
ReSpec version: 34.2.0
File a bug: https://github.com/w3c/respec/
Error: undefined

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20w3c/html-aam%23510.)._
</details>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -14979,7 +15031,61 @@ <h4 id=att-step>`step`</h4>
...
+    <tr>
+      <th>Comments</th>
+      <td>
+        The `switch` attribute modifies the semantics and behavior of the `input` element in the checkbox state.
+        See <a href="#el-input-checkbox-switch">`input type=checkbox switch`</a> for mapping details.
+      </td>
+    </tr>
+  </tbody>
+</table>
+<h4 id=att-switch>`style`</h4>
```
Comment (scottaohara) [inline index.html]:
```suggestion
<h4 id=att-style>`style`</h4>
```

**Review by annevk — APPROVED** [review]
Since there's a dedicated entry I assume that the indeterminate state of `input` elements ends up being always ignored here? In the implementation I instead had to modify the `aria-checked` computation, but I think this works too.

**Comment by scottaohara** [comment]
logged this bug today per the current webkit implementation that uses pseudo content to derive a name for the switch when it is not labelled - https://bugs.webkit.org/show_bug.cgi?id=270377

i'm going to wait for response on that before i make any other updates to this PR.  But, definitely not a useful name with the current implementation.

related ARIA issue where maybe the custom state presented as pseudo content could be exposed in a different/better way - https://github.com/w3c/aria/issues/2085#issuecomment-1973438719

**Comment by rahimabdi** [comment]
Closing, this PR has been migrated to the ARIA monorepo:
- https://github.com/w3c/aria/pull/2217

**Pull request closed without merging**